diff -up linux-5.15/drivers/nvme/host/core.c.12~ linux-5.15/drivers/nvme/host/core.c
--- linux-5.15/drivers/nvme/host/core.c.12~	2023-04-30 15:23:54.302410008 +0200
+++ linux-5.15/drivers/nvme/host/core.c	2023-04-30 15:26:12.430602533 +0200
@@ -2543,7 +2543,16 @@ static const struct nvme_core_quirk_entr
 		.quirks = NVME_QUIRK_DELAY_BEFORE_CHK_RDY |
 			  NVME_QUIRK_NO_DEEPEST_PS |
 			  NVME_QUIRK_IGNORE_DEV_SUBNQN,
-	}
+	},
+        {
+		/* https://forum.openmandriva.org/t/nvme-ssd-m2-not-seen-by-omlx-4-0/2407
+		 * Unknown SSD ... Maybe ADATA/Hynix (a similar mn from ADTA
+		 * but vid seems to be Hynix)
+		 */
+		.vid = 0x1c5c,
+		.mn = "HFM128GDHTNG-8310B",
+		.quirks = NVME_QUIRK_NO_APST,
+ 	}
 };
 
 /* match is null-terminated but idstr is space-padded. */
